internal/controller/controllers/agent_controller.go

```diff
@@ -80,69 +81,255 @@ func (r *AgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+		}
 	}
-	return result, nil
+	conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+		Type:    aiv1beta1.SpecSyncedCondition,
+		Status:  condStatus,
+		Reason:  reason,
+		Message: msg,
+	})
 }
 
-func (r *AgentReconciler) updateInventory(c *common.Cluster, agent *aiv1beta1.Agent) error {
+func installed(agent *aiv1beta1.Agent, status, statusInfo string) {
+	var condStatus corev1.ConditionStatus
+	var reason string
+	var msg string
+	switch status {
+	case models.HostStatusInstalled:
+		condStatus = corev1.ConditionTrue
+		reason = aiv1beta1.AgentInstalledReason
+		msg = fmt.Sprintf("%s %s", aiv1beta1.AgentInstalledMsg, statusInfo)
+	case models.HostStatusError:
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.AgentInstallationFailedReason
+		msg = fmt.Sprintf("%s %s", aiv1beta1.AgentInstallationFailedMsg, statusInfo)
+	case models.HostStatusInsufficient, models.HostStatusDisconnected, models.HostStatusDiscovering,
+		models.HostStatusPendingForInput, models.HostStatusKnown:
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.AgentInstallationNotStartedReason
+		msg = aiv1beta1.AgentInstallationNotStartedMsg
+	case models.HostStatusPreparingForInstallation, models.HostStatusPreparingSuccessful,
+		models.HostStatusInstalling, models.HostStatusInstallingInProgress,
+		models.HostStatusInstallingPendingUserAction:
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.AgentInstallationInProgressReason
+		msg = fmt.Sprintf("%s %s", aiv1beta1.AgentInstallationInProgressMsg, statusInfo)
+	default:
+		condStatus = corev1.ConditionUnknown
+		reason = aiv1beta1.UnknownStatusReason
+		msg = fmt.Sprintf("%s %s", aiv1beta1.UnknownStatusMsg, status)
+	}
+	conditionsv1.SetStatusCondition(&agent.Status.Conditions, conditionsv1.Condition{
+		Type:    aiv1beta1.InstalledCondition,
+		Status:  condStatus,
+		Reason:  reason,
+		Message: msg,
+	})
+}
 
-	host := getHostFromCluster(c, agent.Name)
-	if host == nil {
-		r.Log.Errorf("Fail to update inventory: Host %s not found in cluster %s", agent.Name, c.Name)
-		return errors.New("Fail to update inventory: Host not found in cluster")
+func validated(agent *aiv1beta1.Agent, status string, h *models.Host) {
+	failedValidationInfo := ""
+	validationRes, err := host.GetValidations(h)
+	var failures []string
+	if err == nil {
+		for _, vRes := range validationRes {
+			for _, v := range vRes {
+				if v.Status == host.ValidationFailure {
+					failures = append(failures, v.Message)
+				}
+			}
+		}
+		failedValidationInfo = strings.Join(failures[:], ",")
+	}
+	var condStatus corev1.ConditionStatus
+	var reason string
+	var msg string
+	switch status {
+	case models.HostStatusInsufficient:
+		condStatus = corev1.ConditionFalse
+		reason = aiv1beta1.AgentValidationsFailingReason
+		msg = fmt.Sprintf("%s %s", aiv1beta1.AgentValidationsFailingMsg, failedValidationInfo)
+	default:
+		if h.ValidationsInfo == "" {
```

Comment:
This could be a good opportunity for an argument-less switch.

```
switch {
  case status = models.HostStatusInsufficient:
    // stuff
  case h.ValidationsInfo == "":
    // other stuff
  default:
    // default stuff
}
```

Reply:
Done